sql/plan/alter_table.go

```diff
@@ -491,6 +491,12 @@ func (i *addColumnIter) rewriteTable(ctx *sql.Context, rwt sql.RewritableTable)
 
 	rowIter := sql.NewTableRowIter(ctx, rwt, partitions)
 
+	var autoTbl sql.AutoIncrementTable
+	if newSch.HasAutoIncrement() {
+		autoTbl = rwt.(sql.AutoIncrementTable)
```

Comment:
This will panic if the interface doesn't match, always use the two-param version